aristanetworks/ansible-cvp
# Add support for reordering of configlet in cv_device_v3

## Change Summary
This PR adds the functionnality to be able to reorder configlets applied to devices in the cv_device_v3 module. 
Example: 
If the following configlets are applied to a device in that order: 

- A
- B 
- C 

The following playbook can be used in order to change the order to (C - B - A) : 
```
- hosts: cv_server
  vars:
    devices: 
      - fqdn: DC1-SPINE1
        parentContainerName: DC1_SPINES
        configlets:
          - C
          - B
          - A
  tasks:
    - name: "Re-order configlet device"
      arista.cvp.cv_device_v3:
        devices: "{{devices}}"
        state: present          
 ```

A few point to keep in mind while using this: 
* This is a beta feature, please report here if any misbehavior is noticed
* The re-order configlets will be added to the end of the list of configlets. 

Example: If the following configlets are applied to the device: 
```
- A
- B 
- Base_configlet
- C
```
And if the above playbook is used (note that the configlet `Base_configlet` is missing from the playbook), the resulting order will be: 
```
- Base_configlet
- C 
- B
- A
```

## Types of changes


## Related Issue(s)

Fixes #362

## Component(s) name

`arista.cvp.cv_device_v3`

## Proposed changes
This PR adds the functionnality to reorder configlets in the cv_device_v3 module. 
The order defined in the playbook will be the one applied to the device. 


## How to test
There is no new syntax for this change however, this requires a change being made in the cvprac package (as of today). 
In order to test this functionality, please do the following:
Step 1: Git clone the cvprac github repo
```
git clone https://github.com/aristanetworks/cvprac
```

Step 2: Load the cvprac package - recommended using this inside a virtual environment or a container.
```
pip install -e cvprac
```

Step 3: Verification - the cvprac package used should have the version `develop`
```
pip list
[...]
cvprac              develop   /projects/cvprac
[...]
```

## Checklist:
